File: react_native/HFilmTrainingApp/src/datas/Repository.js
import FilmRepository from './film/FilmRepository'
import FilmApiDataSource from "./film/FilmApiDataSource";
import UserRepository from "./user/UserRepository";

export default class Repository {

  constructor() {
    Repository._film = null;
    Repository._user = null;
  }

  static get film() {
    if (this._film != null) return this._film;
    this._film = new FilmRepository(new FilmApiDataSource());
    return this._film;
  }

  static get user() {
    if (this._user != null) return this._user;
    this._user = new UserRepository();
    return this._user;
  }
}

File: android/app/src/main/java/com/phamnhuvu/android/hfilmtrainingapp/datas/Repository.kt
package com.phamnhuvu.android.hfilmtrainingapp.datas

import com.phamnhuvu.android.hfilmtrainingapp.datas.film.FilmApiDataSource
import com.phamnhuvu.android.hfilmtrainingapp.datas.film.FilmLocalDataSource
import com.phamnhuvu.android.hfilmtrainingapp.datas.film.FilmRepository
import com.phamnhuvu.android.hfilmtrainingapp.datas.user.UserApiDataSource
import com.phamnhuvu.android.hfilmtrainingapp.datas.user.UserRepository

class Repository {

  companion object {
    var Film: FilmRepository? = null
      get() {
        if (field != null) return field
        field = FilmRepository(
          FilmApiDataSource(),
          FilmLocalDataSource())
        return field
      }

    var User: UserRepository? = null
      get() {
        if (field != null) return field
        field = UserRepository(
          UserApiDataSource())
        return field
      }
  }
}